src/crunker.ts

```diff
@@ -295,24 +295,32 @@ export default class Crunker {
    *
    * @internal
    */
-  private _writeHeaders(buffer: Float32Array): DataView {
-    const arrayBuffer = new ArrayBuffer(44 + buffer.length * 2);
+  private _writeHeaders(buffer: Float32Array, numOfChannels: number, sampleRate: number, isFloat32: boolean): DataView {
+    const format = isFloat32 ? 3 : 1;
+    const bitDepth = isFloat32 ? 32 : 16;
+    const bytesPerSample = bitDepth / 8;
+    const sampleSize = numOfChannels * bytesPerSample;
+
+    const arrayBuffer = new ArrayBuffer(44 + buffer.length * bytesPerSample);
     const view = new DataView(arrayBuffer);
 
     this._writeString(view, 0, 'RIFF');
-    view.setUint32(4, 32 + buffer.length * 2, true);
+    view.setUint32(4, 36 + buffer.length * bytesPerSample, true);
     this._writeString(view, 8, 'WAVE');
     this._writeString(view, 12, 'fmt ');
     view.setUint32(16, 16, true);
-    view.setUint16(20, 1, true);
-    view.setUint16(22, 2, true);
-    view.setUint32(24, this._sampleRate, true);
-    view.setUint32(28, this._sampleRate * 4, true);
-    view.setUint16(32, 4, true);
-    view.setUint16(34, 16, true);
+    view.setUint16(20, format, true);
+    view.setUint16(22, numOfChannels, true);
+    view.setUint32(24, sampleRate, true);
+    view.setUint32(28, sampleRate * sampleSize, true);
+    view.setUint16(32, sampleSize, true);
+    view.setUint16(34, bitDepth, true);
     this._writeString(view, 36, 'data');
-    view.setUint32(40, buffer.length * 2, true);
+    view.setUint32(40, buffer.length * bytesPerSample, true);
 
+    if (isFloat32) {
+      return this._copyFloat32(view, buffer, 44);
+    }
     return this._floatTo16BitPCM(view, buffer, 44);
   }
```

Comment:
`suggestion-non-blocking`: Let's be explicit here to demostrate there are no alterntive code paths expected. E.g it's an "exhaustive check".
```suggestion
    if (isFloat32) {
      return this._copyFloat32(view, buffer, 44);
    } else {
      return this._floatTo16BitPCM(view, buffer, 44);
    }
  }
```